code/modules/mob/living/living.dm

```diff
@@ -1366,9 +1366,9 @@
  * * stacks: int The amount to set fire_stacks to
  */
 /mob/living/proc/set_fire_stacks(stacks)
-	fire_stacks = clamp(stacks, -20, 20)
-	if(fire_stacks <= 0)
+	if(stacks <= 0)
 		extinguish_mob()
+	fire_stacks = clamp(stacks, -20, 20)
```

Comment:
This makes it so decreasing your fire stacks below 0 won't automatically extinguish the mob. Can you make sure this doesn't break something?